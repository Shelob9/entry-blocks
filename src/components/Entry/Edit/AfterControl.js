import React, {Fragment} from 'react';
import {TextControl} from '@wordpress/components'
import {componentClassName} from "../../../control-factory/componentClassName";

export default function (props) {
	const {
		after,
		setAfter,
		instanceId,
		hideLabel
	} = props;

	const label = 'Text After Value';
	const className = componentClassName('entryValue', 'after', 'edit')
	const id = className + '-' + instanceId;

	if( hideLabel ){
		return(
			<Fragment>
				<label
					className="sr-only screen-reader-text"
					htmlFor={id}
				>
					{ label }
				</label>
				<input
					type="text"
					value={after}
					placeholder={'Text to show after value'}
					onChange={(event) => setAfter(event.target.value)}
					id={id}
				/>
			</Fragment>
		)
	}


	return (
		<TextControl
			className={className}
			label={label}
			id={id}
			value={after}
			onChange={setAfter}
		/>
	);
}


