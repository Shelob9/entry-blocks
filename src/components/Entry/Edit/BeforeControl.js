import React, {Fragment} from 'react';
import {TextControl} from '@wordpress/components'
import {componentClassName} from "../../../control-factory/componentClassName";

export default function (props) {
	const {
		before,
		setBefore,
		hideLabel,
		instanceId
	} = props;

	const label = 'Text Before Value';
	const className = componentClassName('entryValue', 'before', 'edit');
	const id = className + '-' + instanceId;

	if( hideLabel ){
		return(
			<Fragment>
				<label
					className="sr-only screen-reader-text"
					htmlFor={id}
				>
					{ label }
				</label>
				<input
					type="text"
					value={before}
					placeholder={'Text to show before value'}
					onChange={(event) => setBefore(event.target.value)}
				/>
			</Fragment>
		)
	}

	return (
		<TextControl
			className={className}
			id={id}
			label={label}
			value={before}
			onChange={setBefore}
		/>
	);
}


