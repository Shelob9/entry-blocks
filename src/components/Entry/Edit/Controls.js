import React, {createElement, Fragment} from 'react';
import {findFormById} from "../findFormById";

import {ChooseForm} from "../../controls/ChooseForm";
import {ChooseEntryField} from "../../controls/ChooseEntryField";
import {ChooseEntry} from "../../controls/ChooseEntry";
import BeforeControl from '../../../components/Entry/Edit/BeforeControl';
import AfterControl from '../../../components/Entry/Edit/AfterControl';

export default function (props)  {
	const {
		entries,
		formId,
		entryId,
		onSetForm,
		onSetEntry,
		instanceId,
		onSetField,
		entryFieldId,
		before,
		after,
		setBefore,
		setAfter,
		forms
	} = props;
	let form = {};
	if (!props.form) {
		form = findFormById(forms, formId);
	} else {
		form = props.form;
	}

	const elements = [
		<ChooseForm
			forms={forms}
			currentFormId={formId}
			onSetForm={onSetForm}
			instanceId={instanceId}
			key={1}
		/>
	];

	if (formId) {
		elements.push(<ChooseEntry
			entries={entries}
			currentEntry={entryId}
			onSetEntry={onSetEntry}
			instanceId={instanceId}
			key={2}
		/>);
	}

	if (entryId) {
		elements.push(<ChooseEntryField
			currentEntry={entryId}
			form={form}
			onSetField={onSetField}
			instanceId={instanceId}
			entries={entries}
			entryFieldId={entryFieldId}
			key={3}

		/>);
		elements.push(
			<BeforeControl
				setBefore={setBefore}
				before={before}
				instanceId={instanceId}
				key={5}
			/>
		);
		elements.push(
			<AfterControl
				setAfter={setAfter}
				after={after}
				instanceId={instanceId}
				key={6}
			/>
		);
	}

	return createElement(Fragment, {}, elements);
}
