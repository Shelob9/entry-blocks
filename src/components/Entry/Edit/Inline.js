import React, {Fragment} from 'react';

import {ChooseForm} from "../../controls/ChooseForm";
import {ChooseEntry} from "../../controls/ChooseEntry";
import {ChooseEntryField} from "../../controls/ChooseEntryField";
import BeforeControl from './BeforeControl';
import AfterControl from './AfterControl';


export function EntryValueInlineEditor(props) {
	const {
		before,
		after,
		setAfter,
		setBefore,
		instanceId,
		getChooseEntryField
	} = props;

	return (
		<Fragment>
				<span>
					<BeforeControl
						setBefore={setBefore}
						before={before}
						instanceId={instanceId}
						hideLabel={true}
					/>
				</span>

				<span>
					{getChooseEntryField()}
				</span>

			<span>
				<AfterControl
					setAfter={setAfter}
					after={after}
					instanceId={instanceId}
					hideLabel={true}

				/>
			</span>

		</Fragment>
	);
}


export default function (props) {
	const {
		forms,
		entries,
		formId,
		entryId,
		onSetForm,
		onSetEntry,
		instanceId,
		onSetField,
		entryFieldId,
		before,
		after,
		setBefore,
		setAfter
	} = props;
	if ('undefined' === typeof forms) {
		return <Fragment/>
	}

	const form = forms.find(form => formId === form.ID);
	if (!formId) {
		return <ChooseForm
			forms={forms}
			currentFormId={formId}
			onSetForm={onSetForm}
			instanceId={instanceId}
		/>
	}

	if (!entryId) {
		return <ChooseEntry
			entries={entries}
			currentEntry={entryId}
			onSetEntry={onSetEntry}
			instanceId={instanceId}
		/>
	}

	let entryField = null;

	function getChooseEntryField() {
		if (props.ChooseEntryField) {
			return props.ChooseEntryField;
		}
		const hideLabel = entryFieldId && entryFieldId.length;
		return <ChooseEntryField
			currentEntry={entryId}
			form={form}
			onSetField={onSetField}
			instanceId={instanceId}
			entries={entries}
			entryFieldId={entryFieldId}
			hideLabel={hideLabel}
		/>;
	}

	if (entryId && !entryFieldId) {

		const entry = entries.hasOwnProperty(entryId) ? entries[entryId] : null;
		if (entry) {
			entryField = entry.fields[entryFieldId];

		}
		return (
			getChooseEntryField()
		)

	}


	if (entryFieldId) {
		return EntryValueInlineEditor({
			...props,
			getChooseEntryField
		});

	}


}


