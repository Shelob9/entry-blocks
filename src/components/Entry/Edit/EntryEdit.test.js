import "@babel/polyfill";


import renderer from 'react-test-renderer';
import React from 'react';
import Enzyme, {shallow,mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

import AfterControl from './AfterControl';
import BeforeControl from "./BeforeControl";

import Inline from './Inline';
import Controls from './Controls';

const _forms = require( '../../../__MOCK_DATA__/forms' )
const forms = Object.values(_forms );
const formId = 'CF5bde22b646ed0';
const form = _forms[formId];
const entries = require( '../../../__MOCK_DATA__/entries' );
const entry = entries[0];
const entryFieldId = 'fld_8768091';
const entryId = entry.id;
const instanceId = '54fds';
const beforeText = 'Before';
const afterText = 'After';
const onSetForm = jest.fn();
const onSetEntry = () => jest.fn();
const onSetEntryField =jest.fn();
const setBefore = () => jest.fn();
const setAfter = () => jest.fn();

describe( 'AfterControl', ()  => {
	test( 'After editor with label hidden matches spapshots',() => {
		expect( renderer.create(
			<AfterControl
				setAfter={setAfter}
				after={afterText}
				instanceId={instanceId}
				hideLabel={true}
			/>
		)).toMatchSnapshot();
	});

	test( 'After editor with label visible matches spapshots',() => {
		expect( renderer.create(
			<AfterControl
				setAfter={setAfter}
				after={afterText}
				instanceId={instanceId}
				hideLabel={false}
			/>
		)).toMatchSnapshot();
	});

	test( 'After editor with label visible calls onChange',() => {

		const mockFn = jest.fn();
		const event = {
			preventDefault() {},
			target: { value: 'matched' }
		};
		const component = mount(
			<AfterControl
				setAfter={mockFn}
				after={afterText}
				instanceId={instanceId}
				hideLabel={false}
			/>);
		component.find('input').simulate('change', event);
		expect(mockFn.mock.calls[0][0]).toBe('matched');

	});

	test( 'After editor with label not visible calls onChange',() => {

		const mockFn = jest.fn();
		const event = {
			preventDefault() {},
			target: { value: 'matched' }
		};
		const component = mount(
			<AfterControl
				setAfter={mockFn}
				after={afterText}
				instanceId={instanceId}
				hideLabel={true}
			/>);
		component.find('input').simulate('change', event);
		expect(mockFn.mock.calls[0][0]).toBe('matched');

	});

});

describe( 'Before editor', ()  => {
	test( 'Before editor with label hidden matches spapshots',() => {
		expect( renderer.create(
			<BeforeControl
				setBefore={setBefore}
				before={beforeText}
				instanceId={instanceId}
				hideLabel={true}
			/>
		)).toMatchSnapshot();
	});

	test( 'Before editor with label visible matches spapshots',() => {
		expect( renderer.create(
			<BeforeControl
				setBefore={setBefore}
				before={beforeText}
				instanceId={instanceId}
				hideLabel={true}
			/>
		)).toMatchSnapshot();
	});

	test( 'Before editor with label visible calls onChange',() => {

		const mockFn = jest.fn();
		const event = {
			preventDefault() {},
			target: { value: 'matched' }
		};
		const component = mount(
			<BeforeControl
				setBefore={mockFn}
				before={beforeText}
				instanceId={instanceId}
				hideLabel={true}
			/>);
		component.find('input').simulate('change', event);
		expect(mockFn.mock.calls[0][0]).toBe('matched');

	});

	test( 'Before editor with label not visible calls onChange',() => {

		const mockFn = jest.fn();
		const event = {
			preventDefault() {},
			target: { value: 'matched' }
		};
		const component = mount(
			<BeforeControl
				setBefore={mockFn}
				before={beforeText}
				instanceId={instanceId}
				hideLabel={true}
			/>);
		component.find('input').simulate('change', event);
		expect(mockFn.mock.calls[0][0]).toBe('matched');

	});
});


describe( 'Inline component', ()  => {
	const mockFn = jest.fn();

	it( 'Match snapshot with just forms prop set',() => {
		expect(
			renderer.create(
				<Inline
					entryId={''}
					entries={{}}
					onSetEntry={mockFn}
					formId={''}
					forms={forms}
					form={{}}
					onSetForm={mockFn}
					instanceId={instanceId}
					onSetField={mockFn}
					entryFieldId={entryFieldId}
					before={''}
					after={''}
					setBefore={mockFn}
					setAfter={mockFn}

				/>
			)
		).toMatchSnapshot();
	} );

	it( 'Match snapshot with form selected only',() => {
		expect(
			renderer.create(
				<Inline
					entryId={''}
					entries={{}}
					onSetEntry={mockFn}
					formId={formId}
					forms={forms}
					form={form}
					onSetForm={mockFn}
					instanceId={instanceId}
					onSetField={mockFn}
					entryFieldId={''}
					before={''}
					after={''}
					setBefore={mockFn}
					setAfter={mockFn}

				/>
			)
		).toMatchSnapshot();
	} );

	it( 'Match snapshot with form and entry selected only',() => {
		expect(
			renderer.create(
				<Inline
					entryId={entryId}
					entries={entries}
					onSetEntry={mockFn}
					formId={formId}
					forms={forms}
					form={form}
					onSetForm={mockFn}
					instanceId={instanceId}
					onSetField={mockFn}
					entryFieldId={''}
					before={''}
					after={''}
					setBefore={mockFn}
					setAfter={mockFn}

				/>
			)
		).toMatchSnapshot();

	});

	it( 'Match snapshot with all props set',() => {
		expect(
			renderer.create(
				<Inline
					entryId={entryId}
					entries={entries}
					onSetEntry={mockFn}
					formId={formId}
					forms={forms}
					form={form}
					onSetForm={mockFn}
					instanceId={instanceId}
					onSetField={mockFn}
					entryFieldId={entryFieldId}
					before={beforeText}
					after={afterText}
					setBefore={mockFn}
					setAfter={mockFn}

				/>
			)
		).toMatchSnapshot();
	} );


	it( 'Match snapshot with render prop for ChooseEntryField passed',() => {
		expect(
			renderer.create(
				<Inline
					entryId={entryId}
					entries={entries}
					onSetEntry={mockFn}
					formId={formId}
					forms={forms}
					form={form}
					onSetForm={mockFn}
					instanceId={instanceId}
					onSetField={mockFn}
					entryFieldId={entryFieldId}
					before={beforeText}
					after={afterText}
					setBefore={mockFn}
					setAfter={mockFn}
					ChooseEntryField={<div>Hi Roy</div>}

				/>
			)
		).toMatchSnapshot();
	} );
});
describe( 'Controls component', ()  => {
	const mockFn = jest.fn();

	it( 'Match snapshot with just forms prop set',() => {
		expect(
			renderer.create(
				<Controls
					entryId={''}
					entries={{}}
					onSetEntry={mockFn}
					formId={''}
					forms={forms}
					form={{}}
					onSetForm={mockFn}
					instanceId={instanceId}
					onSetField={mockFn}
					entryFieldId={entryFieldId}
					before={''}
					after={''}
					setBefore={mockFn}
					setAfter={mockFn}

				/>
			)
		).toMatchSnapshot();
	} );

	it( 'Match snapshot with form selected only',() => {
		expect(
			renderer.create(
				<Controls
					entryId={''}
					entries={{}}
					onSetEntry={mockFn}
					formId={formId}
					forms={forms}
					form={form}
					onSetForm={mockFn}
					instanceId={instanceId}
					onSetField={mockFn}
					entryFieldId={''}
					before={''}
					after={''}
					setBefore={mockFn}
					setAfter={mockFn}

				/>
			)
		).toMatchSnapshot();
	} );

	it( 'Match snapshot with form and entry selected only',() => {
		expect(
			renderer.create(
				<Controls
					entryId={entryId}
					entries={entries}
					onSetEntry={mockFn}
					formId={formId}
					forms={forms}
					form={form}
					onSetForm={mockFn}
					instanceId={instanceId}
					onSetField={mockFn}
					entryFieldId={''}
					before={''}
					after={''}
					setBefore={mockFn}
					setAfter={mockFn}

				/>
			)
		).toMatchSnapshot();

	});

	it( 'Match snapshot with all props set',() => {
		expect(
			renderer.create(
				<Controls
					entryId={entryId}
					entries={entries}
					onSetEntry={mockFn}
					formId={formId}
					forms={forms}
					form={form}
					onSetForm={mockFn}
					instanceId={instanceId}
					onSetField={mockFn}
					entryFieldId={entryFieldId}
					before={beforeText}
					after={afterText}
					setBefore={mockFn}
					setAfter={mockFn}

				/>
			)
		).toMatchSnapshot();
	} );
});

