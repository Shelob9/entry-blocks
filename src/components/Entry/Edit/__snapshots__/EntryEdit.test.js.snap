exports[`AfterControl After editor with label hidden matches spapshots 1`] = `
Array [
  <label
    className="sr-only screen-reader-text"
    htmlFor="caldera-after-entryValue-edit-54fds">
    Text After Value
</label>,
  <input
    id="caldera-after-entryValue-edit-54fds"
    onChange={[Function]}
    placeholder="Text to show after value"
    type="text"
    value="After" />,
]
`;

exports[`AfterControl After editor with label visible matches spapshots 1`] = `
<div
  className="components-base-control caldera-after-entryValue-edit">
  <div
    className="components-base-control__field">
    <label
      className="components-base-control__label"
      htmlFor="inspector-text-control-0">
      Text After Value
    </label>
    <input
      aria-describedby={undefined}
      className="components-text-control__input"
      id="caldera-after-entryValue-edit-54fds"
      onChange={[Function]}
      type="text"
      value="After" />
  </div>
</div>
`;

exports[`Before editor Before editor with label hidden matches spapshots 1`] = `
Array [
  <label
    className="sr-only screen-reader-text"
    htmlFor="caldera-before-entryValue-edit-54fds">
    Text Before Value
</label>,
  <input
    onChange={[Function]}
    placeholder="Text to show before value"
    type="text"
    value="Before" />,
]
`;

exports[`Before editor Before editor with label visible matches spapshots 1`] = `
Array [
  <label
    className="sr-only screen-reader-text"
    htmlFor="caldera-before-entryValue-edit-54fds">
    Text Before Value
</label>,
  <input
    onChange={[Function]}
    placeholder="Text to show before value"
    type="text"
    value="Before" />,
]
`;

exports[`Controls component Match snapshot with all props set 1`] = `
Array [
  <div>
    <label
        className=""
        htmlFor="caldera-forms-form-chooser-54fds">
        Choose Form
    </label>
    <select
        className="caldera-forms-form-chooser"
        id="caldera-forms-form-chooser-54fds"
        onChange={[Function]}
        value="CF5bde22b646ed0">
        <option>
             -- 
        </option>
        <option
            value="CF5bde22a843265">
            Not Contact Form
        </option>
        <option
            value="CF5bde22b646ed0">
            Contact Form
        </option>
    </select>
</div>,
  <label
    htmlFor="caldera-forms-entry-chooser-54fds">
    Choose Entry
</label>,
  <select
    className="caldera-forms-entry-chooser"
    id="caldera-forms-entry-chooser-54fds"
    onChange={[Function]}
    value="1">
    <option
        value={null}>
        --
    </option>
    <option
        value="1">
        1
    </option>
    <option
        value="2">
        2
    </option>
</select>,
  <label
    className=""
    htmlFor="caldera-forms-entry-field-chooser-54fds">
    Choose Field
</label>,
  <select
    className="caldera-forms-entry-field-chooser"
    id="caldera-forms-entry-field-chooser-54fds"
    onChange={[Function]}
    value="fld_8768091">
    <option
        value={null}>
        --
    </option>
    <option
        value="id">
        ID
    </option>
    <option
        value="datestamp">
        Submitted
    </option>
    <option
        value="fld_8768091">
        First Name
    </option>
    <option
        value="fld_9970286">
        Last Name
    </option>
    <option
        value="fld_6009157">
        Email Address
    </option>
    <option
        value="fld_7683514">
        Comments / Questions
    </option>
</select>,
  <div
    className="components-base-control caldera-before-entryValue-edit">
    <div
        className="components-base-control__field">
        <label
            className="components-base-control__label"
            htmlFor="inspector-text-control-4">
            Text Before Value
        </label>
        <input
            aria-describedby={undefined}
            className="components-text-control__input"
            id="caldera-before-entryValue-edit-54fds"
            onChange={[Function]}
            type="text"
            value="Before" />
    </div>
</div>,
  <div
    className="components-base-control caldera-after-entryValue-edit">
    <div
        className="components-base-control__field">
        <label
            className="components-base-control__label"
            htmlFor="inspector-text-control-5">
            Text After Value
        </label>
        <input
            aria-describedby={undefined}
            className="components-text-control__input"
            id="caldera-after-entryValue-edit-54fds"
            onChange={[Function]}
            type="text"
            value="After" />
    </div>
</div>,
]
`;

exports[`Controls component Match snapshot with form and entry selected only 1`] = `
Array [
  <div>
    <label
        className=""
        htmlFor="caldera-forms-form-chooser-54fds">
        Choose Form
    </label>
    <select
        className="caldera-forms-form-chooser"
        id="caldera-forms-form-chooser-54fds"
        onChange={[Function]}
        value="CF5bde22b646ed0">
        <option>
             -- 
        </option>
        <option
            value="CF5bde22a843265">
            Not Contact Form
        </option>
        <option
            value="CF5bde22b646ed0">
            Contact Form
        </option>
    </select>
</div>,
  <label
    htmlFor="caldera-forms-entry-chooser-54fds">
    Choose Entry
</label>,
  <select
    className="caldera-forms-entry-chooser"
    id="caldera-forms-entry-chooser-54fds"
    onChange={[Function]}
    value="1">
    <option
        value={null}>
        --
    </option>
    <option
        value="1">
        1
    </option>
    <option
        value="2">
        2
    </option>
</select>,
  <label
    className=""
    htmlFor="caldera-forms-entry-field-chooser-54fds">
    Choose Field
</label>,
  <select
    className="caldera-forms-entry-field-chooser"
    id="caldera-forms-entry-field-chooser-54fds"
    onChange={[Function]}
    value="">
    <option
        value={null}>
        --
    </option>
    <option
        value="id">
        ID
    </option>
    <option
        value="datestamp">
        Submitted
    </option>
    <option
        value="fld_8768091">
        First Name
    </option>
    <option
        value="fld_9970286">
        Last Name
    </option>
    <option
        value="fld_6009157">
        Email Address
    </option>
    <option
        value="fld_7683514">
        Comments / Questions
    </option>
</select>,
  <div
    className="components-base-control caldera-before-entryValue-edit">
    <div
        className="components-base-control__field">
        <label
            className="components-base-control__label"
            htmlFor="inspector-text-control-2">
            Text Before Value
        </label>
        <input
            aria-describedby={undefined}
            className="components-text-control__input"
            id="caldera-before-entryValue-edit-54fds"
            onChange={[Function]}
            type="text"
            value="" />
    </div>
</div>,
  <div
    className="components-base-control caldera-after-entryValue-edit">
    <div
        className="components-base-control__field">
        <label
            className="components-base-control__label"
            htmlFor="inspector-text-control-3">
            Text After Value
        </label>
        <input
            aria-describedby={undefined}
            className="components-text-control__input"
            id="caldera-after-entryValue-edit-54fds"
            onChange={[Function]}
            type="text"
            value="" />
    </div>
</div>,
]
`;

exports[`Controls component Match snapshot with form selected only 1`] = `
Array [
  <div>
    <label
        className=""
        htmlFor="caldera-forms-form-chooser-54fds">
        Choose Form
    </label>
    <select
        className="caldera-forms-form-chooser"
        id="caldera-forms-form-chooser-54fds"
        onChange={[Function]}
        value="CF5bde22b646ed0">
        <option>
             -- 
        </option>
        <option
            value="CF5bde22a843265">
            Not Contact Form
        </option>
        <option
            value="CF5bde22b646ed0">
            Contact Form
        </option>
    </select>
</div>,
  <label
    htmlFor="caldera-forms-entry-chooser-54fds">
    Choose Entry
</label>,
  <select
    className="caldera-forms-entry-chooser"
    id="caldera-forms-entry-chooser-54fds"
    onChange={[Function]}
    value="">
    <option
        value={null}>
        --
    </option>
</select>,
]
`;

exports[`Controls component Match snapshot with just forms prop set 1`] = `
<div>
  <label
    className=""
    htmlFor="caldera-forms-form-chooser-54fds">
    Choose Form
  </label>
  <select
    className="caldera-forms-form-chooser"
    id="caldera-forms-form-chooser-54fds"
    onChange={[Function]}
    value="">
    <option>
       -- 
    </option>
    <option
      value="CF5bde22a843265">
      Not Contact Form
    </option>
    <option
      value="CF5bde22b646ed0">
      Contact Form
    </option>
  </select>
</div>
`;

exports[`Inline component Match snapshot with all props set 1`] = `
Array [
  <span>
    <label
        className="sr-only screen-reader-text"
        htmlFor="caldera-before-entryValue-edit-54fds">
        Text Before Value
    </label>
    <input
        onChange={[Function]}
        placeholder="Text to show before value"
        type="text"
        value="Before" />
</span>,
  <span>
    <label
        className="sr-only screen-reader-text"
        htmlFor="caldera-forms-entry-field-chooser-54fds">
        Choose Field
    </label>
    <select
        className="caldera-forms-entry-field-chooser"
        id="caldera-forms-entry-field-chooser-54fds"
        onChange={[Function]}
        value="fld_8768091">
        <option
            value={null}>
            --
        </option>
        <option
            value="id">
            ID
        </option>
        <option
            value="datestamp">
            Submitted
        </option>
        <option
            value="fld_8768091">
            First Name
        </option>
        <option
            value="fld_9970286">
            Last Name
        </option>
        <option
            value="fld_6009157">
            Email Address
        </option>
        <option
            value="fld_7683514">
            Comments / Questions
        </option>
    </select>
</span>,
  <span>
    <label
        className="sr-only screen-reader-text"
        htmlFor="caldera-after-entryValue-edit-54fds">
        Text After Value
    </label>
    <input
        id="caldera-after-entryValue-edit-54fds"
        onChange={[Function]}
        placeholder="Text to show after value"
        type="text"
        value="After" />
</span>,
]
`;

exports[`Inline component Match snapshot with form and entry selected only 1`] = `
Array [
  <label
    className=""
    htmlFor="caldera-forms-entry-field-chooser-54fds">
    Choose Field
</label>,
  <select
    className="caldera-forms-entry-field-chooser"
    id="caldera-forms-entry-field-chooser-54fds"
    onChange={[Function]}
    value="">
    <option
        value={null}>
        --
    </option>
    <option
        value="id">
        ID
    </option>
    <option
        value="datestamp">
        Submitted
    </option>
    <option
        value="fld_8768091">
        First Name
    </option>
    <option
        value="fld_9970286">
        Last Name
    </option>
    <option
        value="fld_6009157">
        Email Address
    </option>
    <option
        value="fld_7683514">
        Comments / Questions
    </option>
</select>,
]
`;

exports[`Inline component Match snapshot with form selected only 1`] = `
Array [
  <label
    htmlFor="caldera-forms-entry-chooser-54fds">
    Choose Entry
</label>,
  <select
    className="caldera-forms-entry-chooser"
    id="caldera-forms-entry-chooser-54fds"
    onChange={[Function]}
    value="">
    <option
        value={null}>
        --
    </option>
</select>,
]
`;

exports[`Inline component Match snapshot with just forms prop set 1`] = `
<div>
  <label
    className=""
    htmlFor="caldera-forms-form-chooser-54fds">
    Choose Form
  </label>
  <select
    className="caldera-forms-form-chooser"
    id="caldera-forms-form-chooser-54fds"
    onChange={[Function]}
    value="">
    <option>
       -- 
    </option>
    <option
      value="CF5bde22a843265">
      Not Contact Form
    </option>
    <option
      value="CF5bde22b646ed0">
      Contact Form
    </option>
  </select>
</div>
`;

exports[`Inline component Match snapshot with render prop for ChooseEntryField passed 1`] = `
Array [
  <span>
    <label
        className="sr-only screen-reader-text"
        htmlFor="caldera-before-entryValue-edit-54fds">
        Text Before Value
    </label>
    <input
        onChange={[Function]}
        placeholder="Text to show before value"
        type="text"
        value="Before" />
</span>,
  <span>
    <div>
        Hi Roy
    </div>
</span>,
  <span>
    <label
        className="sr-only screen-reader-text"
        htmlFor="caldera-after-entryValue-edit-54fds">
        Text After Value
    </label>
    <input
        id="caldera-after-entryValue-edit-54fds"
        onChange={[Function]}
        placeholder="Text to show after value"
        type="text"
        value="After" />
</span>,
]
`;
